### PR TITLE
setup redux

### PR DESCRIPTION
**Pull request by VelzckC0D3**
# Bookstore -  `Redux` `React` `ES6` `API`

## ✨ _Requested Implementations_ ✨

✔️ _Added Redux Toolkid using the provided npm command_
✔️ Added a directory that contains all the redux logic `/src/redux`_
✔️ _Configured a Redux store `/src/redux/store.js`_
✔️ _Defined a `slice` of state for `books` that Includes an empty array of books and also a reducer for `add` & `remove` books_
✔️ _Defined a `slice` of state for `categories` that includes an empty array, also a reducer that `checks the status and returns "under construction"_
✔️ _Structured with the ducks pattern_
✔️ _Checked all Linters_

#  💡 _Thank you for your review!_ 💡

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  categories: [],
+  status: 'Under construction',
+};
+
+export const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    // Here I'll add the reducers for categories
+  },
```
Comment (Ibizugbe) [inline src/redux/categories/categoriesSlice.js]: Nice job so far👍. Just as required for this milestone, we are required to add a reducer that checks the status😉 Kindly see the image below and make the necessary change👍.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @VelzckC0D3 👋👋👋, 

tryagain


Good job so far!👍
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are properly setup and they are passing👍.
- [x] GItflow have been used👍.
- [x] `@reduxjs/toolkit` & `react-redux` are present in `package.json`👍.
- [x] The structure of the files follows the duck pattern👍.
- [x] An initial state have been craeted for the books with two reducers `addbook` and `removeBook` 👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [OPTIONAL] Kindly ensure to capitalize your commit messages subsequently as this act is more professional👍.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (VelzckC0D3) [reply src/redux/categories/categoriesSlice.js]: @Ibizugbe thank you! I didn't noticed that I didn't add it, corrected now !

**Review by Munsa1 — APPROVED** [review]
Hi @VelzckC0D3, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 
Good Communication :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**